### PR TITLE
Clear Algolia Index Before Doing a Full Index.

### PR DESCRIPTION
**Pull request by erikkgagewhitt**
### What does this PR do, or why is it needed?

Clears the Algolia index before attempting a full index. We do this because, while additions are always caught by the full index, any content that gets deleted will not actually be removed from the index.

### How do I test this PR?

I suggest you test this once in master and once on this branch.

1. Create a new piece of content in ROCK and publish it
1. Uncomment ln 11 in `apollos-data-connector-algolia`
2. Start your server, waiting for the index to finish
3. Search for your content, you should find it
4. Delete your content
5. Run the index again, waiting for it to finish
6. Search for your content again; if you're on this branch, you should not find it

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by vinnyjth** [comment]
🙌

File: packages/apollos-data-connector-algolia-search/src/data-source.js [hunk 1]
```diff
@@ -57,6 +57,11 @@ query getItem {
   }
 
   async indexAll() {
+    this.index.clearIndex((err) => {
```
Comment (vinnyjth) [inline packages/apollos-data-connector-algolia-search/src/data-source.js]: @erikkgagewhitt We might want we can promisify this and then `await` the result. I'm not sure what will happen if the index is still clearing when we try and index more items.
something like
```
await new Promise((resolve, reject) => {
 this.index.clearIndex((err) => {
  if (err) {reject(err)}
  resolve()
 })
})

Comment (erikkgagewhitt) [reply packages/apollos-data-connector-algolia-search/src/data-source.js]: I was under the impression that Algolia queued requests, but this is probably way safer. I'll add that in!

Comment (erikkgagewhitt) [reply packages/apollos-data-connector-algolia-search/src/data-source.js]: All set!
